## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 438399 --checkout commit`
Commit: `29901b4b5271cf7eb36ced3201c6973dc6a68fd2`

---
### `x86_64-linux`
<details>
  <summary>:x: 1 package failed to build:</summary>
  <ul>
    <li>rocmPackages.hiprt (rocmPackages_6.hiprt)</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 62 packages built:</summary>
  <ul>
    <li>adaptivecpp</li>
    <li>adaptivecppWithCuda</li>
    <li>adaptivecppWithRocm</li>
    <li>blender-hip</li>
    <li>llama-cpp-rocm</li>
    <li>magma (magma-hip)</li>
    <li>magma.test (magma-hip.test)</li>
    <li>ollama-rocm</li>
    <li>python312Packages.torchWithRocm</li>
    <li>python312Packages.torchWithRocm.cxxdev</li>
    <li>python312Packages.torchWithRocm.dev</li>
    <li>python312Packages.torchWithRocm.dist</li>
    <li>python312Packages.torchWithRocm.lib</li>
    <li>python313Packages.torchWithRocm</li>
    <li>python313Packages.torchWithRocm.cxxdev</li>
    <li>python313Packages.torchWithRocm.dev</li>
    <li>python313Packages.torchWithRocm.dist</li>
    <li>python313Packages.torchWithRocm.lib</li>
    <li>rocmPackages.aotriton (rocmPackages_6.aotriton)</li>
    <li>rocmPackages.clr (rocmPackages_6.clr)</li>
    <li>rocmPackages.clr.icd (rocmPackages_6.clr.icd)</li>
    <li>rocmPackages.composable_kernel (rocmPackages_6.composable_kernel)</li>
    <li>rocmPackages.hipblas (rocmPackages_6.hipblas)</li>
    <li>rocmPackages.hipblaslt (rocmPackages_6.hipblaslt)</li>
    <li>rocmPackages.hipcub (rocmPackages_6.hipcub)</li>
    <li>rocmPackages.hipfft (rocmPackages_6.hipfft)</li>
    <li>rocmPackages.hiprand (rocmPackages_6.hiprand)</li>
    <li>rocmPackages.hipsolver (rocmPackages_6.hipsolver)</li>
    <li>rocmPackages.hipsparse (rocmPackages_6.hipsparse)</li>
    <li>rocmPackages.migraphx (rocmPackages_6.migraphx)</li>
    <li>rocmPackages.miopen (rocmPackages.miopen-hip, rocmPackages_6.miopen, rocmPackages_6.miopen-hip)</li>
    <li>rocmPackages.mivisionx (rocmPackages.mivisionx-hip, rocmPackages_6.mivisionx, rocmPackages_6.mivisionx-hip)</li>
    <li>rocmPackages.mivisionx-cpu (rocmPackages_6.mivisionx-cpu)</li>
    <li>rocmPackages.mpi (rocmPackages.openmpi, rocmPackages_6.mpi, rocmPackages_6.openmpi)</li>
    <li>rocmPackages.mpi.dev (rocmPackages.openmpi.dev, rocmPackages_6.mpi.dev, rocmPackages_6.openmpi.dev)</li>
    <li>rocmPackages.mpi.man (rocmPackages.openmpi.man, rocmPackages_6.mpi.man, rocmPackages_6.openmpi.man)</li>
    <li>rocmPackages.mscclpp (rocmPackages_6.mscclpp)</li>
    <li>rocmPackages.rccl (rocmPackages_6.rccl)</li>
    <li>rocmPackages.rccl-tests (rocmPackages_6.rccl-tests)</li>
    <li>rocmPackages.rccl-tests.test (rocmPackages_6.rccl-tests.test)</li>
    <li>rocmPackages.rocalution (rocmPackages_6.rocalution)</li>
    <li>rocmPackages.rocblas (rocmPackages_6.rocblas)</li>
    <li>rocmPackages.rocfft (rocmPackages_6.rocfft)</li>
    <li>rocmPackages.rocm-tests (rocmPackages_6.rocm-tests)</li>
    <li>rocmPackages.rocmPath (rocmPackages_6.rocmPath)</li>
    <li>rocmPackages.rocminfo (rocmPackages_6.rocminfo)</li>
    <li>rocmPackages.rocmlir (rocmPackages.rocmlir-rock, rocmPackages_6.rocmlir, rocmPackages_6.rocmlir-rock)</li>
    <li>rocmPackages.rocprim (rocmPackages_6.rocprim)</li>
    <li>rocmPackages.rocprofiler (rocmPackages_6.rocprofiler)</li>
    <li>rocmPackages.rocprofiler-register (rocmPackages_6.rocprofiler-register)</li>
    <li>rocmPackages.rocr-debug-agent (rocmPackages_6.rocr-debug-agent)</li>
    <li>rocmPackages.rocrand (rocmPackages_6.rocrand)</li>
    <li>rocmPackages.rocsolver (rocmPackages_6.rocsolver)</li>
    <li>rocmPackages.rocsparse (rocmPackages_6.rocsparse)</li>
    <li>rocmPackages.rocthrust (rocmPackages_6.rocthrust)</li>
    <li>rocmPackages.roctracer (rocmPackages_6.roctracer)</li>
    <li>rocmPackages.rocwmma (rocmPackages_6.rocwmma)</li>
    <li>rocmPackages.rpp (rocmPackages.rpp-hip, rocmPackages_6.rpp, rocmPackages_6.rpp-hip)</li>
    <li>rocmPackages.rpp-cpu (rocmPackages_6.rpp-cpu)</li>
    <li>rocmPackages.tensile (rocmPackages_6.tensile)</li>
    <li>rocmPackages.tensile.dist (rocmPackages_6.tensile.dist)</li>
    <li>zluda</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `x86_64-linux`</summary>
<details>
<summary>rocmPackages.hiprt</summary>
<pre>      |                    ^
/build/source/contrib/Orochi/ParallelPrimitives/RadixSort.h:133:15: note: move assignment operator of 'RadixSort' is implicitly deleted because field 'm_oroutils' is of reference type 'OrochiUtils &'
  133 |         OrochiUtils& m_oroutils;
      |                      ^
/build/source/contrib/Orochi/ParallelPrimitives/RadixSort.h:62:49: note: replace 'default' with 'delete'
   62 |         RadixSort& operator=( RadixSort&& ) noexcept = default;
      |                                                        ^~~~~~~
      |                                                        delete
1 warning and 10 errors generated.
make[2]: *** [CMakeFiles/hiprt02005.dir/build.make:135: CMakeFiles/hiprt02005.dir/hiprt/impl/Compiler.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
1 warning generated.
8 warnings generated.
8 warnings generated.
3 warnings generated.
1 warning generated.
1 warning generated.
8 warnings generated.
make[1]: *** [CMakeFiles/Makefile2:90: CMakeFiles/hiprt02005.dir/all] Error 2
make: *** [Makefile:136: all] Error 2</pre>
</details>
</details>

--- a/pkgs/development/python-modules/torch/source/default.nix
+++ b/pkgs/development/python-modules/torch/source/default.nix
@@ -1,7 +1,6 @@
 {
   stdenv,
   lib,
-  pkgs,
   fetchFromGitHub,
   fetchFromGitLab,
   fetchpatch2,
@@ -251,9 +250,6 @@ let
     "Magma cudaPackages does not match cudaPackages" =
       cudaSupport
       && (effectiveMagma.cudaPackages.cudaMajorMinorVersion != cudaPackages.cudaMajorMinorVersion);
-    # Should be fixed by WIP https://github.com/NixOS/nixpkgs/pull/438399
-    "ROCm support on non-default python versions is temporarily broken" =
-      rocmSupport && (pkgs.python3.version != python.version);
   };
 
   unroll-src = writeShellScript "unroll-src" ''

--- a/pkgs/development/rocm-modules/6/hiprt/default.nix
+++ b/pkgs/development/rocm-modules/6/hiprt/default.nix
@@ -5,6 +5,7 @@
   cmake,
   clr,
   gcc,
+  python3,
 }:
 
 stdenv.mkDerivation (finalAttrs: {
@@ -25,6 +26,7 @@ stdenv.mkDerivation (finalAttrs: {
   nativeBuildInputs = [
     gcc # required for replacing easy-encryption binary
     cmake
+    python3
   ];
 
   buildInputs = [

--- a/pkgs/development/rocm-modules/6/rocminfo/default.nix
+++ b/pkgs/development/rocm-modules/6/rocminfo/default.nix
@@ -30,13 +30,15 @@ stdenv.mkDerivation (finalAttrs: {
     sha256 = "sha256-fQPtO5TNbCbaZZ7VtGkkqng5QZ+FcScdh1opWr5YkLU=";
   };
 
+  strictDeps = true;
+
   nativeBuildInputs = [
     cmake
     rocm-cmake
+    python3
   ];
 
   buildInputs = [ rocm-runtime ];
-  propagatedBuildInputs = [ python3 ];
   cmakeFlags = [ "-DROCRTST_BLD_TYPE=Release" ];
 
   prePatch = ''

